src/Lib.hs

```diff
@@ -44,17 +47,115 @@ startPetSelection user goldRemaining = do
 startBattle :: User -> IO ()
 startBattle user = do
     putStrBar
-    putStrLn $ show (userRoster user)
-    putStrLn "\nBATTLE"
-    putStrLn "BATTLE"
-    putStrLn "BATTLE"
-    -- TODO BATTLE
+    putStrLn "YOUR TEAM"
+    putStrLn $ display (userRoster user)
+    -- putStrLn $ show (userRoster user)
+    
+    -- Create opponent team
+    petOpponent1 <- getPet allPets
+    petOpponent2 <- getPet allPets
+    petOpponent3 <- getPet allPets
+    let 
+      opponentRoster = Roster
+        { rosterPet1 = Just petOpponent1
+        , rosterPet2 = Just petOpponent2
+        , rosterPet3 = Just petOpponent3
+        , rosterPet4 = Nothing
+        , rosterPet5 = Nothing
+        , rosterPet6 = Nothing
+        }
+    let
+      userOpponent = User
+        { userName = "OPPONENT"
+        , userRoster = opponentRoster
+        , userItemList = itemListEmpty
+        }
+    putStrLn "\nOPPONENT TEAM"
+    putStrLn $ show (opponentRoster)
+
+    -- BATTLE
+    battleRoster user userOpponent
+
+
+battleRoster :: User -> User -> IO ()
+battleRoster user1 user2 = do
+  let r1pet = getRosterFirst (userRoster user1)
+  let r2pet = getRosterFirst (userRoster user2)
+
+  when (isNothing r1pet) (putStrLn "user1 LOSES")
+  when (isNothing r2pet) (putStrLn "user2 LOSES")
+
+  let user1Pet = fromJust r1pet
+  let user2Pet = fromJust r2pet
+  putStrLn "\n"
+  putStrLn $ userName user1 ++ ": " ++ display user1Pet
+  putStrLn $ "vs."
+  putStrLn $ userName user2 ++ ": " ++ display user2Pet
+
+  -- battle pets
+  let (user1Pet', user2Pet') = battlePets user1Pet user2Pet
```

Comment:
So this sort of looks like what I just mentioned above, except that it is pure so it excludes printing above, it would be nice to keep this function but also make a function that calls battle pets but also does IO.